yellowbrick/features/manifold.py

```diff
@@ -299,179 +361,103 @@ def fit_transform(self, X, y=None):
 
         Note also that fit records the amount of time it takes to fit the
         manifold and reports that information in the visualization.
-
+        
         Parameters
         ----------
         X : array-like of shape (n, m)
-            A matrix or data frame with n instances and m features where m > 2.
+            A matrix or data frame with n instances and m features
 
         y : array-like of shape (n,), optional
             A vector or series with target values for each instance in X. This
             vector is used to determine the color of the points in X.
 
         Returns
         -------
-        self : Manifold
-            Returns the visualizer object.
+        Xprime : array-like of shape (n, 2)
+            Returns the 2-dimensional embedding of the instances.
+        
         """
-        # Determine target type
-        self._determine_target_color_type(y)
-
-        # Compute classes and colors if target type is discrete
-        if self._target_color_type == DISCRETE:
-            self.classes_ = np.unique(y)
-
-            color_kwargs = {'n_colors': len(self.classes_)}
-
-            if isinstance(self.colors, str):
-                color_kwargs['colormap'] = self.colors
-            else:
-                color_kwargs['colors'] = self.colors
-
-            self._colors = resolve_colors(**color_kwargs)
-
-        # Compute target range if colors are continuous
-        elif self._target_color_type == CONTINUOUS:
-            y = np.asarray(y)
-            self.range_ = (y.min(), y.max())
-
+        # Call to fit determines color target type and features.
+        super(Manifold, self).fit(X, y, **kwargs)
         with Timer() as self.fit_time_:
             Xp = self.manifold.fit_transform(X)
-
         self.draw(Xp, y)
         return Xp
 
-    def transform(self, X):
+    def transform(self, X, y=None, **kwargs):
         """
         Returns the transformed data points from the manifold embedding.
 
         Parameters
         ----------
         X : array-like of shape (n, m)
             A matrix or data frame with n instances and m features
-
+            
+        y : array-like of shape (n,), optional
+            The target, used to specify the colors of the points.
+            
         Returns
         -------
         Xprime : array-like of shape (n, 2)
             Returns the 2-dimensional embedding of the instances.
+            
+        Note
+        ----
+        This method does not work with MDS, TSNE and SpectralEmbedding because 
+        it is yet to be implemented in sklearn.
         """
         try:
-            return self.manifold.transform(X)
+            Xp = self.manifold.transform(X)
```

Comment:
Sorry I didn't catch this before - but shouldn't this return `Xp`?

Reply:
`transform`  is supposed to be the entry point of the `draw` method. And it returns Xp after the `draw` call!

Reply:
But this raises yet another error! When `transform` is called followed by `fit`, `draw` is called two times. So, to counter this we need to remove `draw` call from `transform`. Hence it will be a tradeoff between these two.
To keep these both we can add conditionals based on the type of Manifold.
What do you suggest? Any other methods that can make this work?